llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -141,13 +143,26 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorShiftsByZero(
 bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
     MachineInstr &MI) const {
   const Register DstReg = MI.getOperand(0).getReg();
-  MachineInstr *ExtOp = MRI.getUniqueVRegDef(MI.getOperand(2).getReg());
+  MachineInstr *ExtOp = getDefIgnoringCopies(MI.getOperand(2).getReg(), MRI);
 
-  assert(ExtOp && "Expected SSA.");
-  if (ExtOp->getOpcode() != AIE2::G_INTRINSIC ||
-      cast<GIntrinsic>(*ExtOp).getIntrinsicID() !=
-          Intrinsic::aie2_extract_I128_I512)
+  if (!isa<GIntrinsic>(MI) || !isa<GIntrinsic>(*ExtOp))
     return false;
+  switch (cast<GIntrinsic>(MI).getIntrinsicID()) {
+  case Intrinsic::aie2_set_I512_I128: {
+    if (cast<GIntrinsic>(*ExtOp).getIntrinsicID() !=
+        Intrinsic::aie2_extract_I128_I512)
+      return false;
+    break;
+  }
+  case Intrinsic::aie2_set_I512_I256: {
+    if (cast<GIntrinsic>(*ExtOp).getIntrinsicID() !=
+        Intrinsic::aie2_ext_I256_I512)
+      return false;
+    break;
+  }
+  default:
+    return false;
+  }
```

Comment:
You seem to be "fixing" what was just defined in the previous commit. Maybe have all the changes for `tryToCombineSetExtract` in a single commit?

Reply:
(And add relevant test cases for the newly supported `aie2_ext_I256_I512`)